sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2839,11 +2852,22 @@ pi_int32 ExecCGCommand::enqueueImpQueue() {
                backend::ext_intel_esimd_emulator);
         if (MEvent != nullptr)
           MEvent->setHostEnqueueTime();
-        MQueue->getPlugin()->call<PiApiKind::piEnqueueKernelLaunch>(
-            nullptr,
-            reinterpret_cast<pi_kernel>(ExecKernel->MHostKernel->getPtr()),
-            NDRDesc.Dims, &NDRDesc.GlobalOffset[0], &NDRDesc.GlobalSize[0],
-            &NDRDesc.LocalSize[0], 0, nullptr, nullptr);
+        if (ExecKernel->MKernelIsCooperative) {
+          MQueue->getPlugin()
+              ->call<PiApiKind::piextEnqueueCooperativeKernelLaunch>(
+                  nullptr,
+                  reinterpret_cast<pi_kernel>(
+                      ExecKernel->MHostKernel->getPtr()),
+                  NDRDesc.Dims, &NDRDesc.GlobalOffset[0],
+                  &NDRDesc.GlobalSize[0], &NDRDesc.LocalSize[0], 0, nullptr,
+                  nullptr);
+        } else {
+          MQueue->getPlugin()->call<PiApiKind::piEnqueueKernelLaunch>(
+              nullptr,
+              reinterpret_cast<pi_kernel>(ExecKernel->MHostKernel->getPtr()),
+              NDRDesc.Dims, &NDRDesc.GlobalOffset[0], &NDRDesc.GlobalSize[0],
+              &NDRDesc.LocalSize[0], 0, nullptr, nullptr);
```

Comment:
There are a lot of arguments here. I'd like to inline c-style comments  for what the literals are e.g. `/* foo */0, /* bar */ nullptr`